docs/raw/organisation-channels.md

```diff
@@ -0,0 +1,160 @@
+`permissions` is a list of permissions set by the creator of the organisation
+`chats` is a list of chats/(channels?) associated with the organisation
+
+
+### Permissions
+
+Permissions can be set at the organisation level or at the chat level.
+
+Permissions at the chat level cannot override permissions at the organisation level.
+(Here I will describe rules, but most of them should be simple, i.e you can't set a
+channel public in an private organisation, you can have a member of a chat who is not
+a member of an organisation)
+
+`ens_only` means that only user who have an `ENS` can join the organisation
+`private` indicates whether the organisation channels are encrypted with a secret
+key shared only by members, otherwise the key is derivable by the chat_id
+`access` indicates how membership works for this organisation. 
+When is `NO_MEMBERSHIP` anyone can read and write on the channels, `private` MUST be set to false in this case.
+When is `INVITE_ONLY` users MUST NOT request access, if `private` is set to `false` the channel
+will be read-only. Only upon receiving an `OrganisationInvitation` a user will be able to join.
+When is `ON_REQUEST` users MUST send a `OrganisationRequestJoin` to join the organisation.
+
+
+## Joining an organisation
+
+If an organisation access is set to `ON_REQUEST` a user MUST send a
+message `OrganisationRequestJoin` if they want to join the organisation.
+A `chat_id` MAY also be specified if they are interested in joining a particular channel
+of the organisation that has access also set to `ON_REQUEST`.
+
+An organisation MUST send back a `OrganisationRequestJoinResponse` with the `accepted`
+flag set.
+If set to `true` a `grant` MUST be also sent which is constructed by:
+
+```
+   Signature(public-key,org-id,clock/some other stuff)
+```
+
+```
+This part I am not sure, but I think some kind of mechanism is needed, 
+the correct way would be to piggy back the whole description of the org, but 
+that's likely to resources intensive. So a grant is basically the next best thing, but
+there might be other methods.
+```
+
+This grant MAY be attached to each message so that client can optimistically process
+messages coming from this user even though the most up-to-date `OrganistationDescription` has not been fetched.
+
+(The idea here is that we don't want to have issue with ordering of messages, for example
+if I joined channel `x`, and I fetch messages from the mailserver, I might see a post from a new
+  user `y` before I have fetched the org description, so I don't know whether the user 
+  is actually allowed to post. By using a grant, I can optimistically process their messages,
+  knowing that at some point between my latest version of the memberships and what I haven't fetched
+  yet, they have been added to the org. In case they have been removed afterward, I can go back and delete those 
+  messages, but in most cases I should be fine.
```

Comment:
I like this solution, I have a few questions.

- When will new user `y` know to stop sending messages with a `grant`?
- Because a grant is more available than an `OrganisationDescription`, is the disadvantage of using a `grant` all the time the additional payload to the community message over the single source `OrganisationDescription`? ie why not always use grants?
- Do `grants` expire?

Reply:
> When will new user y know to stop sending messages with a grant?

There's no easy way to know, so I would have the client keep grant as long as it's not invalid, but say 7 days should be plenty for now for example.

> Because a grant is more available than an OrganisationDescription, is the disadvantage of using a grant all the time the additional payload to the community message over the single source OrganisationDescription? ie why not always use grants?

The issue with grants is that they are not completely authoritative and that they only reflect partial updates, so clients should only optimistically act on them, and revert if necessary.
If we were to always use grants, then it would be easy to have inconsistent state among clients (Which is always possible, but we should minimize it). 

So you could see a `OrganisationDescription` as a checkpoint, and grants as patches essentially. Eventually we want this to be consolidated in a single `OrganisationdDescription`, which will "squash" the history, so that we don't waste too much bandwidth.

It is essentially a mixture between a purely event driven protocol (private group chats), which is very bandwidth heavy but has little loss of information/room for state inconsistencies,  and a purely materialized view, which has high loss of information, room for state inconsistencies, but low bandwidth.

Reply:
Why not sending an `OrganisationDescription` message every time that an update occurs (in addition to 24 hrs period)? This way, as soon as a new member joins, all the clients get informed via the new advertised  `OrganisationDescription`. 
I understand the bandwidth concern, but if the new member is to send multiple messages as soon as she joins, then the additional `grant` information, attached to each message, is already causing an access bandwidth which on aggregate MAY cause more bandwidth consumption than having the organization owner send one single`OrganisationDescription` per member insertion

Reply:
> Why not sending an OrganisationDescription message every time that an update occurs

Sorry, it's not specified, but that's the implemented behavior (on top of resending), will update the docs

We'd still need the grant I think in case clients have partial history, as the admin might not always be online and it's possible they are not able to retrieve historic messages.